Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,416 @@
+        0x58DD00,            // 25 Wanted
+        0x58DD7E,            // 26 Wanted
+        0x58DF71,            // 27 Wanted
+        0x58DFE5,            // 28 Wanted
+        0x58B09F,            // 29 Vehicle names
+        0x58B13F,            // 30 Vehicle names x pos
+        0x58AD3A,            // 31 Area names scale
+        0x58AD65,            // 32 Area names ?
+        0x58AE4A,            // 33 Area names x pos
+        0x58C395,            // 34 Subs
+        0x58C41D,            // 35 Subs
+        0x58C4DC,            // 36 Subs
+        0x4E9F30,            // 37 RadioStation
+    };
+
+    uintptr_t hudHeightAddresses[34] = {
+        0x58EB29,            // 0 Clock
+        0x58EBF9,            // 1 Clock
+        0x58F546,            // 2 Money
+        0x58F5CE,            // 3 Money
+        0x589346,            // 4 Info bars
+        0x58EE60,            // 5 Info bars
+        0x588B9C,            // 6 Info bars
+        0x58EEC8,            // 7 Info bars
+        0x58913E,            // 8 Info bars
+        0x58EF32,            // 9 Info bars
+        0x58EF99,            // 10 Info bars
+        0x589216,            // 11 Info bars
+        0x58F0F8,            // 12 Info bars
+        0x58F168,            // 13 Info bars
+        0x58D945,            // 14 Weapon icons
+        0x58D882,            // 15 Weapon icons
+        0x58F90B,            // 16 Weapon icons
+        0x5894AF,            // 17 Ammo
+        0x58F9C0,            // 18 Ammo y position
+        0x58FA4A,            // 19 Ammo
+        0x58DCA2,            // 20 Wanted
+        0x58DD68,            // 21 Wanted
+        0x58DDF4,            // 22 Wanted
+        0x58DF55,            // 23 Wanted
+        0x58DF9B,            // 24 Wanted
+        0x58DEE4,            // 25 Wanted
+        0x58B089,            // 26 Vehicle names
+        0x58B12D,            // 27 Vehicle names
+        0x58AD24,            // 28 Area names
+        0x58AE0D,            // 29 Area names
+        0x58C37F,            // 30 Subs
+        0x58C407,            // 31 Subs
+        0x58C4C6,            // 32 Subs
+        0x4E9F1A,            // 33 RadioStation
+    };
+
+    for (size_t i = 0; i < std::size(hudWidthAddresses); i++)
+    {
+        MemPut<float*>(hudWidthAddresses[i] + 0x2, &HUDWidth[i]);
+    }
+
+    for (size_t i = 0; i < std::size(hudHeightAddresses); i++)
+    {
+        MemPut<float*>(hudHeightAddresses[i] + 0x2, &HUDHeight[i]);
+    }
+
+    // Second player fix.
+    MemPut<float*>(0x58F9A0 + 0x2, &HUDWidth[38]);             // Weapon icon X
+    MemPut<float*>(0x58F993 + 0x2, &HUDWidth[16]);             // Weapon icon X
+    MemPut<float*>(0x58F972 + 0x2, &HUDHeight[16]);            // Weapon icon Y
+    MemPut<float*>(0x58FA8E + 0x2, &HUDWidth[17]);             // Ammo x
+}
+
+void InstallFieldOfViewFixes()
+{
+    HookInstall(0x6FF410, (DWORD)SetFOV, 5);
+    // Aiming point
+    HookInstall(0x51499E, (DWORD)CalculateAimingPoint, 5);
```

Comment:
This may cause a crash because the JMP instruction takes 5 bytes, there are only 4 bytes at the specified address, which means that the next instruction will be partially replaced with a random opcode (last byte of the offset)
![image](https://github.com/user-attachments/assets/2aae74b8-d004-4e53-aecc-8371ee4fa399)

The hook size should be 6 here

Reply:
0x6FF410 - hook size should be 9 at least or even 10.
Now this hook breaks the second mov instruction
![image](https://github.com/user-attachments/assets/535f38d1-5823-4817-8e45-c410c0fee849)